pkg/kn/commands/route/describe.go

```diff
@@ -43,23 +48,78 @@ func NewRouteDescribeCommand(p *commands.KnParams) *cobra.Command {
 				return err
 			}
 
-			describeRoute, err := client.GetRoute(args[0])
+			route, err := client.GetRoute(args[0])
 			if err != nil {
 				return err
 			}
 
-			printer, err := routeDescribePrintFlags.ToPrinter()
-			if err != nil {
-				return err
+			if machineReadablePrintFlags.OutputFlagSpecified() {
+				printer, err := machineReadablePrintFlags.ToPrinter()
+				if err != nil {
+					return err
+				}
+				return printer.PrintObj(route, cmd.OutOrStdout())
 			}
-			err = printer.PrintObj(describeRoute, cmd.OutOrStdout())
+			printDetails, err := cmd.Flags().GetBool("verbose")
 			if err != nil {
 				return err
 			}
-			return nil
+			return describe(cmd.OutOrStdout(), route, printDetails)
 		},
 	}
-	commands.AddNamespaceFlags(routeDescribeCommand.Flags(), false)
-	routeDescribePrintFlags.AddFlags(routeDescribeCommand)
-	return routeDescribeCommand
+	flags := command.Flags()
+	commands.AddNamespaceFlags(flags, false)
+	machineReadablePrintFlags.AddFlags(command)
+	flags.BoolP("verbose", "v", false, "More output.")
+	return command
+}
+
+func describe(w io.Writer, route *servingv1.Route, printDetails bool) error {
+	dw := printers.NewPrefixWriter(w)
+	commands.WriteMetadata(dw, &route.ObjectMeta, printDetails)
+	dw.WriteAttribute("URL", route.Status.URL.String())
+	writeService(dw, route, printDetails)
+	dw.WriteLine()
+	writeTraffic(dw, route)
+	dw.WriteLine()
+	commands.WriteConditions(dw, route.Status.Conditions, printDetails)
+	if err := dw.Flush(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeService(dw printers.PrefixWriter, route *servingv1.Route, printDetails bool) {
+	for _, owner := range route.ObjectMeta.OwnerReferences {
+		if owner.Kind != "Service" {
```

Comment:
I wonder whether we should still print `Service:` even when there is no reference to a service ? This make it clear, that no service is attached and makes the output stable (i.e. always a `Service:` label).

wdyt ?

Reply:
I think its a good idea, since we're working Services, Revisions and Routes, we should show relative information, if present or absent.